pepper-apis/src/main/java/org/broadinstitute/ddp/model/event/CopyAnswerEventAction.java

```diff
@@ -1,61 +1,74 @@
-package org.broadinstitute.ddp.service;
+package org.broadinstitute.ddp.model.event;
 
 import java.util.List;
 
 import com.typesafe.config.ConfigFactory;
 import org.broadinstitute.ddp.constants.ConfigFile;
 import org.broadinstitute.ddp.db.AnswerDao;
+import org.broadinstitute.ddp.db.dao.JdbiActivityInstance;
 import org.broadinstitute.ddp.db.dao.JdbiLanguageCode;
-import org.broadinstitute.ddp.db.dto.ActivityInstanceDto;
 import org.broadinstitute.ddp.db.dto.EventConfigurationDto;
 import org.broadinstitute.ddp.exception.DDPException;
 import org.broadinstitute.ddp.model.activity.instance.answer.Answer;
 import org.broadinstitute.ddp.model.activity.instance.answer.BoolAnswer;
 import org.broadinstitute.ddp.model.activity.instance.answer.DateAnswer;
 import org.broadinstitute.ddp.model.activity.instance.answer.DateValue;
 import org.broadinstitute.ddp.model.activity.instance.answer.TextAnswer;
-import org.broadinstitute.ddp.model.event.CopyAnswerValueSetterDefinitions;
-import org.broadinstitute.ddp.model.event.ValueSetter;
+import org.broadinstitute.ddp.model.activity.types.EventTriggerType;
+import org.broadinstitute.ddp.pex.PexInterpreter;
 import org.jdbi.v3.core.Handle;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class CopyAnswerService {
-    private static Logger LOG = LoggerFactory.getLogger(CopyAnswerService.class);
-
-    private static volatile CopyAnswerService instance;
+public class CopyAnswerEventAction extends EventAction {
+    private static final Logger LOG = LoggerFactory.getLogger(CopyAnswerEventAction.class);
+    private CopyAnswerTarget copyAnswerTarget;
+    private String copySourceQuestionStableId;
     private AnswerDao answerDao;
 
-    public static CopyAnswerService getInstance() {
-        if (instance == null) {
-            synchronized (CopyAnswerService.class) {
-                if (instance == null) {
-                    instance = new CopyAnswerService();
-                }
-            }
-        }
-        return instance;
+    public CopyAnswerEventAction(EventConfiguration eventConfiguration, EventConfigurationDto dto) {
+        super(eventConfiguration, dto);
+        this.copyAnswerTarget = dto.getCopyAnswerTarget();
+        copySourceQuestionStableId = dto.getCopySourceQuestionStableId();
+        answerDao = AnswerDao.fromSqlConfig(ConfigFactory.load(ConfigFile.SQL_CONFIG_FILE));
     }
 
-    private CopyAnswerService() {
-        this.answerDao = AnswerDao.fromSqlConfig(ConfigFactory.load(ConfigFile.SQL_CONFIG_FILE));
+    @Override
+    public void doAction(PexInterpreter pexInterpreter, Handle handle, EventSignal eventSignal) {
+        // FIXME The Copy Answer Action was originally written to assume that the activity that triggered the copy
```

Comment:
If these fixmes aren't addressed, do we lose functionality?  The plan I have for copying values from one activity instance into another will address these items.  Just want to make sure that without those changes, nothing will break currently.

Reply:
No your changes will remove these fixmes